internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,193 @@
+package daemonset
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	imageWithVersion = "quay.io/testnetworkfunction/debug-partner:latest"
+	timeout          = 5 * time.Minute
+	nodeExporter     = "node-exporter"
+	containerName    = "container-00"
+	debug            = "debug"
+)
+
+// Delete daemon set
+func DeleteDaemonSet(daemonSetName, namespace string) error {
+	logrus.Infof("Deleting daemon set %s", daemonSetName)
+	deletePolicy := metav1.DeletePropagationForeground
+	client := clientsholder.GetClientsHolder().K8sClient
+	err := client.AppsV1().DaemonSets(namespace).Delete(context.TODO(), daemonSetName, metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy})
+	if err != nil {
+		return fmt.Errorf("daemonset %s deletion failed: %w", daemonSetName, err)
+	}
+	for start := time.Now(); time.Since(start) < timeout; {
+		pods, err := client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "test-network-function.com/app=debug"})
+		if err != nil {
+			return fmt.Errorf("failed to get pods, err: %s", err)
+		}
+
+		if len(pods.Items) == 0 {
+			break
+		}
+		time.Sleep(time.Duration(timeout.Minutes()))
```

Comment:
There's a [previous comment](https://github.com/test-network-function/cnf-certification-test/pull/309#discussion_r927568164) from me about this Sleep that you haven't replied/addressed. I'll copy it here:

You shouldn't use the same `timeout` var here, as this is a more a "retry interval" time for sleeping before doing the next iteration, not a timeout. You can call it daemonsetDeletionCheckRetryInteval and make its value to be 2-5 secs as much.

Also, do not use the `.Minutes()` for this time.Sleep. A time.Duration is a nanoseconds value. So here you're passing 5 nanoseconds to the time.Sleep function here, which is pointless.

Reply:
I fixed it.